src/Authenticator/LoginHandler.php

```diff
@@ -491,6 +491,13 @@ public function redirectAfterSuccessfulLogin(): HTTPResponse
         }
         $request->getSession()->clear(static::SESSION_KEY . '.mustLogin');
 
+        // Deactivate sudo mode that was activated in doLogin()
+        $service = $this->getSudoModeService();
+        // Check if the service has a deactivate method, because it is not defined on the interface
+        if (method_exists($service, 'deactivate')) {
+            call_user_func([$service, 'deactivate'], $this->getRequest()->getSession());
```

Comment:
```suggestion
            $service->deactivate($this->getRequest()->getSession());
```

Reply:
`$this->getSudoModeService()` will return a SudoModeServiceInterface and `deactivate() isn't defined on the interface so looks like an error in my IDE

Reply:
That's what the `hasMethod()` call is for. This code won't be executed if the method isn't there, and if the method _is_ there we should just call it in the normal way that methods get called.

Reply:
> That's what the hasMethod() condition is for. This code won't be executed if the method isn't there, and if the method is there we should just call it in the normal way that methods get called.

Yeah I know it won't get called, but like I say there's an annoying red error in my IDE which hasMethod() won't do anything about.  call_use_func() does the same job minus the red error so will leave as is

I've added an AC to this card to do the CMS 6 update

Reply:
I strongly dislike that but since it's subjective and we have no tie breaker on the team I'll move on.